src/main/java/de/tum/www1/orion/connector/ide/build/OrionBuildConnector.kt

```diff
@@ -0,0 +1,124 @@
+import com.intellij.execution.runners.ExecutionUtil
+import com.intellij.openapi.components.ServiceManager
+import de.tum.www1.orion.build.OrionRunConfiguration
+import de.tum.www1.orion.build.OrionSubmitRunConfigurationType
+import de.tum.www1.orion.build.OrionTestParser
+import de.tum.www1.orion.build.instructor.OrionInstructorBuildUtil
+import de.tum.www1.orion.connector.ide.OrionConnector
+import de.tum.www1.orion.dto.BuildError
+import de.tum.www1.orion.dto.BuildLogFileErrorsDTO
+import de.tum.www1.orion.ui.browser.BrowserWebView
+import de.tum.www1.orion.util.cefRouter
+import org.cef.browser.CefBrowser
+import org.cef.browser.CefFrame
+import org.cef.callback.CefQueryCallback
+import org.cef.handler.CefLoadHandlerAdapter
+import org.cef.handler.CefMessageRouterHandlerAdapter
+import java.util.*
+import java.util.stream.Collectors
+
+class OrionBuildConnector(browserWebView: BrowserWebView) : OrionConnector(browserWebView), IOrionBuildConnector {
+    override fun buildAndTestLocally() {
+        ServiceManager.getService(project, OrionInstructorBuildUtil::class.java).runTestsLocally()
+    }
+
+    override fun onBuildStarted(exerciseInstructions: String) {
+        // Only listen to the first execution result
+        val testParser = ServiceManager.getService(project, OrionTestParser::class.java)
+        if (!testParser.isAttachedToProcess) {
+            val runManager = RunManager.getInstance(project)
+            val settings = runManager
+                    .createConfiguration("Build & Test on Artemis Server", OrionSubmitRunConfigurationType::class.java)
+            (settings.configuration as OrionRunConfiguration).triggeredInIDE = false
+            ExecutionUtil.runConfiguration(settings, DefaultRunExecutor.getRunExecutorInstance())
+            testParser.parseTestTreeFrom(exerciseInstructions)
+        }
+    }
+
+    override fun onBuildFinished() {
+        ServiceManager.getService(project, OrionTestParser::class.java).onTestingFinished()
+    }
+
+    override fun onBuildFailed(buildLogsJsonString: String) {
+        val mapType = object : TypeToken<Map<String?, List<BuildError?>?>?>() {}.type
+        val allErrors = JsonParser().parse(buildLogsJsonString).asJsonObject["errors"]
+        val errors = Gson().fromJson<Map<String, List<BuildError>>>(allErrors, mapType)
+        val buildErrors = errors.entries.stream()
+                .map { fileErrors: Map.Entry<String, List<BuildError>> -> BuildLogFileErrorsDTO(fileErrors.key, fileErrors.value) }
+                .collect(Collectors.toList())
+        val testParser = ServiceManager.getService(project, OrionTestParser::class.java)
+        testParser.onCompileError(buildErrors)
+    }
+
+    override fun onTestResult(success: Boolean, testName: String, message: String) {
+        ServiceManager.getService(project, OrionTestParser::class.java).onTestResult(success, testName, message)
+    }
+
+    override fun initializeHandlers() {
```

Comment:
Originally I just wanted it to work so I followed the [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) principle since the other more general approach is much more complicated. After the plugin has been tested then I will refactor it.

Reply:
Since we are under some time pressure, this is fine. But please create an issue from this conversation, if you won't refactor this in this PR

Reply:
I will create an issue after this merges.